packages/client-sdk-web/src/internal/data-client.ts

```diff
@@ -2566,6 +2570,153 @@ export class DataClient<
+      cacheName,
+      convertToB64String(sortedSetName)
+    );
+
+    this.logger.trace(
+      "'sortedSetLength' request result: %s",
+      truncateString(result.toString())
+    );
+    return result;
+  }
+
+  private async sendSortedSetLength(
+    cacheName: string,
+    sortedSetName: string
+  ): Promise<CacheSortedSetLength.Response> {
+    const request = new _SortedSetLengthRequest();
+    request.setSetName(sortedSetName);
+
+    return await new Promise(resolve => {
+      this.clientWrapper.sortedSetLength(
+        request,
+        {
+          ...this.clientMetadataProvider.createClientMetadata(),
+          ...createCallMetadata(cacheName, this.deadlineMillis),
+        },
+        (err, resp) => {
+          if (resp?.getMissing()) {
+            resolve(new CacheSortedSetLength.Miss());
+          } else if (resp?.getFound()) {
+            const len = resp.getFound()?.getLength();
+            if (!len) {
+              resolve(new CacheSortedSetLength.Miss());
+            } else {
+              resolve(new CacheSortedSetLength.Hit(len));
+            }
+          } else {
+            resolve(
+              new CacheSortedSetLength.Error(cacheServiceErrorMapper(err))
+            );
+          }
+        }
+      );
+    });
+  }
+
+  public async sortedSetLengthByScore(
+    cacheName: string,
+    sortedSetName: string,
+    scoreRange?: {
+      minScore?: number;
+      maxScore?: number;
+    }
+  ): Promise<CacheSortedSetLengthByScore.Response> {
+    try {
+      validateCacheName(cacheName);
+      validateSortedSetName(sortedSetName);
+      if (scoreRange)
+        validateSortedSetScores(scoreRange.minScore, scoreRange.maxScore);
+    } catch (err) {
+      return new CacheSortedSetFetch.Error(normalizeSdkError(err as Error));
+    }
+
+    this.logger.trace(
+      "Issuing 'sortedSetLengthByScore' request; minScore: %s, maxScore: %s",
+      scoreRange?.minScore?.toString() ?? 'null',
+      scoreRange?.maxScore?.toString() ?? 'null'
+    );
+
+    const result = await this.sendSortedSetLengthByScore(
+      cacheName,
+      convertToB64String(sortedSetName),
+      scoreRange
+    );
+
+    this.logger.trace(
+      "'sortedSetLengthByScore' request result: %s",
+      truncateString(result.toString())
+    );
+    return result;
+  }
+
+  private async sendSortedSetLengthByScore(
+    cacheName: string,
+    sortedSetName: string,
+    scoreRange?: {
+      minScore?: number;
+      maxScore?: number;
+    }
+  ): Promise<CacheSortedSetLengthByScore.Response> {
+    const request = new _SortedSetLengthByScoreRequest();
+    request.setSetName(sortedSetName);
+
+    if (scoreRange?.minScore) {
+      request.setInclusiveMin(scoreRange.minScore);
+    } else {
+      request.setUnboundedMin(new _Unbounded());
+    }
+
+    if (scoreRange?.maxScore) {
+      request.setInclusiveMax(scoreRange.maxScore);
+    } else {
+      request.setUnboundedMax(new _Unbounded());
+    }
```

Comment:
Same issue here regarding `undefined` vs `0`. Let's add some tests in the integration tests for these cases as well.